postgis/lwgeom_ogc.c

```diff
@@ -591,26 +591,17 @@ Datum LWGEOM_pointn_linestring(PG_FUNCTION_ARGS)
 PG_FUNCTION_INFO_V1(LWGEOM_x_point);
 Datum LWGEOM_x_point(PG_FUNCTION_ARGS)
 {
-	GSERIALIZED *geom;
-	LWGEOM *lwgeom;
-	LWPOINT *point = NULL;
-	POINT2D p;
-
-	geom = PG_GETARG_GSERIALIZED_P(0);
+	GSERIALIZED *geom = PG_GETARG_GSERIALIZED_P(0);
+	POINT4D pt;
 
-	if ( gserialized_get_type(geom) != POINTTYPE )
+	if (gserialized_get_type(geom) != POINTTYPE)
```

Comment:
Note to self: Maybe this can be moved into the next if condition, and only check it if `gserialized_peek_first_point` fails.

Reply:
what if someone updates gserialized_peek_first_point to peek first point of anything? (I may, it would be a better sorting key than center of bbox)